js/modules/loki_snode_api.js

```diff
@@ -74,40 +98,44 @@ class LokiSnodeAPI {
       }
 
       nodeAddresses.forEach(url => {
-        this.ourSwarmNodes[url] = {};
+        this.ourSwarmNodes[url] = {
+          failureCount: 0,
+        };
       });
     }
     return this.ourSwarmNodes;
   }
 
   async getSwarmNodesByPubkey(pubKey) {
     const swarmNodes = await window.Signal.Data.getSwarmNodesByPubkey(pubKey);
-    // TODO: Check if swarm list is below a threshold rather than empty
-    if (swarmNodes && swarmNodes.size !== 0) {
+    if (swarmNodes) {
       return swarmNodes;
     }
-    return this.replenishSwarm(pubKey);
+    return [];
```

Comment:
if `getSwarmNodesByPubkey` had returned an empty array instead of `null` when there is no conversation associated with the pubkey, then we could have simply done
```js
return window.Signal.Data.getSwarmNodesByPubkey(pubKey);
```
Keeping the return type consistent makes things easier IMO since we know what a function is returning.
Especially when in js, the concept falsey-truthy is quite loose (empty arrays and objects are truthy)

Reply:
If you guys agree, this can be fixed in a separate PR.

Reply:
```suggestion
    return swarmNodes || [];
```
You can remove that if statement and just pass the return like this if we want to keep the empty array as a return